src/elements/walletConnect/WalletConnectButton.tsx

```diff
@@ -1,26 +1,26 @@
 import { UseWalletConnect } from './useWalletConnect';
-import { shortenString } from 'utils/pureFunctions';
-import { Image } from 'components/image/Image';
+import { shortenString, truncateString } from 'utils/pureFunctions';
 import { ReactComponent as IconWallet } from 'assets/icons/wallet.svg';
+import useENS from 'hooks/useENS';
+import Davatar from '@davatar/react';
 
 export const WalletConnectButton = ({
   handleWalletButtonClick,
   account,
   selectedWallet,
 }: UseWalletConnect) => {
-  const buttonText = account ? shortenString(account) : 'Connect Wallet';
+  const { ensName } = useENS(account);
+  const buttonText = account
+    ? truncateString(ensName) || shortenString(account)
+    : 'Connect Wallet';
 
   return (
     <button
       className="flex items-center md:btn-outline-secondary md:btn-sm md:mr-40"
       onClick={() => handleWalletButtonClick()}
     >
       {selectedWallet && account ? (
-        <Image
-          src={selectedWallet.icon}
-          alt="Wallet Logo"
-          className="w-[22px]"
-        />
+        <Davatar size={20} address={account} generatedAvatarType="jazzicon" />
```

Comment:
why do we need this? Why replacing a simple Image with a 3rd party?

Reply:
Davatar is not a using any third-party APIs, they just use eth RPC to fetch avatar from ENS and also show jazzicons if there is no avatar